crates/gpui/src/geometry.rs

```diff
@@ -2831,9 +2837,76 @@ impl From<usize> for Pixels {
     Deserialize,
 )]
 #[repr(transparent)]
-pub struct DevicePixels(pub i32);
+pub struct PhysicalPixels<T>(pub T);
+
+/// Deprecated alias to physical pixels.
+pub type DevicePixels = PhysicalPixels<i32>;
+/// Deprecated alias to physical pixels.
+pub type ScaledPixels = PhysicalPixels<f32>;
+
+/// Constructs a [`PhysicalPixels`] value representing a length in physical pixels.
+///
+/// # Arguments
+///
+/// * `physical_pixels` - The number of physical pixels for the length.
+///
+/// # Returns
+///
+/// A `PhysicalPixels` representing the specified number of pixels.
+pub const fn phypx<T>(physical_pixels: T) -> PhysicalPixels<T> {
+    PhysicalPixels(physical_pixels)
+}
+
+/// Deprecated alias for creating `PhysicalPixels`.
+#[expect(non_snake_case)]
+pub const fn DevicePixels(x: i32) -> DevicePixels {
+    PhysicalPixels(x)
+}
+
+/// Deprecated alias for creating `PhysicalPixels`.
+#[expect(non_snake_case)]
+pub const fn ScaledPixels(x: f32) -> ScaledPixels {
+    PhysicalPixels(x)
+}
+
+impl PhysicalPixels<i32> {
+    /// Creates `PhysicalPixels<i32>` from a `u32` amount of physical pixels.
+    ///
+    /// # Panics
+    ///
+    /// Panics if `cfg!(debug_assertions)` is true and `physical_pixels` is greatrer than
+    /// [`i32::MAX`].
+    pub(crate) fn from_u32(physical_pixels: u32) -> Self {
+        debug_assert!(physical_pixels <= i32::MAX as u32);
+        Self(physical_pixels as _)
+    }
+
+    /// Returns the amount of physical pixels, in `u32`.
+    ///
+    /// # Panics
+    ///
+    /// Panics if `cfg!(debug_assertions)` is true and `self.0` is negative.
+    pub fn as_u32(self) -> u32 {
```

Comment:
Can we change this to something like `as_raw() as u32`? I feel like `as_u32()` is doing a bit too much.

Reply:
Ahhh, I see. Nevermind then :D